### DISCUSSION TITLE
How to delete a cookie once it is expired ?

### DISCUSSION THREAD

**Discussion by Ahtii** [fastapi/fastapi#8960] [Questions] [answered]
Hi
I have created a login endpoint with cookie and jwt token and once the expiry of jwt token has reached I want to remove the cookie from client with `delete_coookie` how would I do ?

**Comment by SirTelemak** [comment, marked as answer]
Best practice is to [SetCookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) with Expires field
This way client need not to ask server for cookie deletion,

But sometimes you want to erase cookies immediately after request (logout, for example), in this case you using header SetCookie again, but fill it with empty value AND set Expires for one second (you could also set time from past for immediate expiration but some browser refuses to set such cookies)

**Comment by eddyizm** [comment]
I believe by default, and good security practice, the jwt token/cookie is `HttpOnly=True` which means the client can't access it via JS. 
FYI - I am using FastAPI to render html templates and just implemented a logout, deleting the cookies from the server side response. As far as I can tell. That's the only way to do it.

**Comment by Frekvens1** [comment]
Stumbled upon the same issue. Sharing my findings as this issue is still marked as open.
`response.delete_cookie("access_token")`

It worked great in my test environment, but not in the production server. Both running in Docker.

A workaround for this issue is as @SirTelemak mentioned, set the cookie with an empty value and set to expire after one second. Posting working code below (You may have to edit to your specific needs):


```
    expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=1)

    response.set_cookie(
        key="access_token",
        value="",
        secure=True,
        httponly=True,
        samesite='none',
        expires=expires.strftime("%a, %d %b %Y %H:%M:%S GMT"),
        domain="example.com"
    )
```

**Reply by Tinitto** [reply]
@Frekvens1 , I also see that in the [starlette set-cookie docs](https://www.starlette.io/responses/#set-cookie), one needs to just set the "max_age" to a negative integer or `0` inorder to discard the cookie immediately. Were you (or anyone else) able to try that as well, and does it work as expected?

Something like:

```python
response.set_cookie(
        key="access_token",
        value="",
        secure=True,
        httponly=True,
        samesite='none',
        expires=1,
        max_age=0, 
        domain="example.com"
    )
```

In my use case, the app redirects to particular URL as long as there is a cookie and I was thinking 1 second might be too long if the app is to redirect to the login page after logout instead of that particular URL...but I have not tested that out yet.
